field_parser.go

```diff
@@ -213,6 +212,16 @@ func (ps *tagBaseFieldParser) ComplementSchema(schema *spec.Schema) error {
 		return fmt.Errorf("invalid type for field: %s", ps.field.Names[0])
 	}
 
+	if IsRefSchema(schema) {
+		oldSchema := schema
+		schema = &spec.Schema{}
+		defer func() {
+			if !reflect.ValueOf(*schema).IsZero() {
+				*oldSchema = *(schema.WithAllOf(*oldSchema))
```

Comment:
Here if the old schema is a $ref one,  a new empty schema is used to receive field properties, and if there are no field properties to set which means the new schema stays empty, then keep the old schema. but if the new schema is not empty,  then wrap the old schema  as a whole one to return.

for example 
if a field `foo` of a type `typeFoo`, which originally is a $ref type of typeBar, 
```
 typeFoo:
    type: object
    properties:
      foo: 
         $ref: '#/definitions/typeBar'
```

if it has some properties to set, such as description，readonly etc.  finally it will be:
```
 typeFoo:
    type: object
    properties:
      foo:
        desription: foo
        readonly: true
        allOf   
          - $ref: '#/definitions/typeBar'
```

Reply:
Ok, I got it.

For readability, you should avoid using defer statements.

Reply:
ok